cmd/kubectl-gadget/utils/trace.go

```diff
@@ -104,169 +134,304 @@ func deleteTraces(contextLogger *log.Entry, traceRestClient *restclient.RESTClie
-	transformLine func(string) string,
-) {
+// getRestClient returns the RESTClient associated with kubeRestConfig() for
+// APIPath /apis and GroupVersion &gadgetv1alpha1.GroupVersion.
+func getRestClient() (*restclient.RESTClient, error) {
+	restConfig, err := kubeRestConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Error while getting rest config: %w", err)
+	}
+
+	traceConfig := *restConfig
+	traceConfig.ContentConfig.GroupVersion = &gadgetv1alpha1.GroupVersion
+	traceConfig.APIPath = "/apis"
+	traceConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
+	traceConfig.UserAgent = restclient.DefaultKubernetesUserAgent()
 
-	contextLogger := log.WithFields(log.Fields{
-		"command": fmt.Sprintf("kubectl-gadget %s", subCommand),
-		"args":    args,
-	})
+	traceRestClient, err := restclient.UnversionedRESTClientFor(&traceConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Error while getting trace rest client: %w", err)
+	}
 
-	traceID := randomTraceID()
+	return traceRestClient, nil
+}
 
+// createTrace creates a trace using Kubernetes REST API.
+func createTrace(trace *gadgetv1alpha1.Trace) error {
 	client, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
 	if err != nil {
-		contextLogger.Fatalf("Error in creating setting up Kubernetes client: %q", err)
+		return fmt.Errorf("Error in creating setting up Kubernetes client: %w", err)
+	}
+
+	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Error in listing nodes: %w", err)
+	}
+
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return fmt.Errorf("Error while getting trace rest client: %w", err)
+	}
+
+	nodeFound := false
+	traceNode := trace.Spec.Node
+
+	for _, node := range nodes.Items {
+		if traceNode != "" && node.Name != traceNode {
+			continue
+		}
+		/*
+		 * If no particular node was given, we need to apply this trace on all
+		 * available nodes.
+		 */
+		if traceNode == "" {
+			trace.Spec.Node = node.Name
+		}
+		nodeFound = true
+
+		err = traceRestClient.
+			Post().
+			Namespace(trace.ObjectMeta.Namespace).
+			Resource("traces").
+			Body(trace).
+			Do(context.TODO()).
+			Error()
+		if err != nil {
+			return fmt.Errorf("Error creating trace on node %q: %w", node.Name, err)
+		}
 	}
 
+	/*
+	 * It is possible we change this value in the above loop, restore it to its
+	 * previous value.
+	 */
+	trace.Spec.Node = traceNode
+
+	if trace.Spec.Node != "" && !nodeFound {
+		return fmt.Errorf("Invalid filter: Node %q does not exist", trace.Spec.Node)
+	}
+
+	return nil
+}
+
+// updateTraceOperation updates operation for an already existing trace using
+// Kubernetes REST API.
+func updateTraceOperation(trace *gadgetv1alpha1.Trace) error {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return fmt.Errorf("Error while getting trace rest client: %w", err)
```

Comment:
This error doesn't add information. The getRestClient function already says exactly that.